libraries/core/src/test/java/com/paritytrading/philadelphia/DoublePipeChannel.java

```diff
@@ -0,0 +1,66 @@
+package com.paritytrading.philadelphia;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ByteChannel;
+import java.nio.channels.GatheringByteChannel;
+import java.nio.channels.Pipe;
+import java.nio.channels.ScatteringByteChannel;
+
+// testing utility - in  prod code this should be selectable, and error handling more robust
+public class DoublePipeChannel implements ByteChannel, ScatteringByteChannel, GatheringByteChannel {
```

Comment:
Could we just call this `PipeChannel`?

I think the comment is not needed either, as this code is in the test scope in any case. To stress that further, let's make the class package-private like all other classes in this package are.